src/bookshelf/src/bksw_wm_base.erl

```diff
@@ -15,28 +15,123 @@
 %% under the License.
 %%
 %% @author Ho-Sheng Hsiao <hosh@chef.io>
-%% @author Tim Dysinger <dysinger@chef.io>
+%% @author Tim Dysinger   <dysinger@chef.io>
+%% @author Lincoln Baker  <lbaker@chef.io>
 
 -module(bksw_wm_base).
 
 %% Complete webmachine callbacks
--export([init/1,
+-export([finish_request/2,
+         init/1,
          is_authorized/2,
-         finish_request/2,
+         malformed_request/2,
          service_available/2]).
 
 %% Helper functions
 -export([create_500_response/2]).
 
+-ifdef(TEST).
+-compile([export_all, nowarn_export_all]).
+-endif.
+
 -include("internal.hrl").
 
 %%
 %% Complete webmachine callbacks
 %%
 
+% why is this here?
+-include_lib("eunit/include/eunit.hrl").
+
+-define(CACHE_CTRL_MAXAGE, "300"  ).
```

Comment:
As discussed we should figure out whether this should be configurable or come in with the request somehow.

Reply:
I investigated this, and compared with what's on master (`bksw_sec.erl`).  Here is my autopsy report.  It appears that the `cache-control` header was originally only set and used with presigned url authentication, not with authorization header authentication.  However, during a past refactor I had apparently lumped `cache-control` in with things common to both presigned url *AND* authorization header authentication.

The fix was to split out the `cache-control` stuff back into presigned_url-exclusive tasks.  Once that was done there was no longer any need to even have an `auth_finish` function so that was eliminated and refactored.  The only difference remaining between what's on master and what I'm doing here is, `cache-control max-age` (how long the link will be considered fresh) needs a number of seconds relative to the time the link is received by the server.  Sigv2 (what's on master) needed to calculate this, since it was more concerned with keeping track of "time on the wall."  With sigv4 we are more concerned with "seconds from now" which is actually what `cache-control max-age` wants, so we just set the `max-age` to something we are already tracking anyway for sigv4 - the amount of seconds until the link expires.